### PR TITLE
feat(tracing): Support basic sampling of transactions

### PR DESCRIPTION
**Pull request by relaxolotl**
Now that transaction contexts exist (#619), determine whether transactions should be sent based on sample rate options. Follows the sampling rules as defined in https://develop.sentry.dev/sdk/performance/#sampling-context.

This is missing some logic related to parent sampling as that property is currently unimplemented on the transaction context.

relates to #601

### CODE REVIEW

**Review by Swatinem — APPROVED** [review]

File: src/sentry_core.c [hunk 1]
```diff
@@ -385,6 +385,45 @@ sentry_capture_event(sentry_value_t event)
     return was_captured ? event_id : sentry_uuid_nil();
 }
 
+bool
+sentry__make_sampling_decision(double sample_rate)
+{
+    uint64_t rnd;
+    return sample_rate < 1.0 && !sentry__getrandom(&rnd, sizeof(rnd))
```
Comment (loewenheim) [inline src/sentry_core.c:392]: Why does this return `false` when `sample_rate ≥ 1.0`? I would interpret that as "everything should be sampled", shouldn't it always return `true` in that case?

Comment (relaxolotl) [reply src/sentry_core.c:392]: thanks for catching this, looks like i messed up that scenario.

Comment (loewenheim) [reply src/sentry_core.c:392]: I wouldn't really expect a method called `is_unsampled` to depend on a random number.

Comment (loewenheim) [reply src/sentry_core.c:392]: I think `is_unsampled` is not a good name either; you'd expect a function with that name to tell you whether a thing passed to it is unsampled, but its only argument is the sampling rate. Really what this function does is return `true` with a probability of `sample_rate`. I would suggest a name like `sentry__random_bool`.

Comment (relaxolotl) [reply src/sentry_core.c]: it is now `sentry__roll_dice` 🎉 this isn't a part of the public API, so this name shouldn't kick up too much of a fuss.

**Pull request merged**